src/main/scala/abalone/variant/Variant.scala

```diff
@@ -102,17 +102,88 @@ abstract class Variant private[variant] (
         )
     }.toMap
 
-  def validSideMoves(@nowarn situation: Situation):  Map[Pos, List[Move]] = {
-    Map()
-  }
+  def validMovesOf2And3(situation: Situation): Map[Pos, List[(String, Move)]] = {
+    def generateMove(orig: Pos, dest: Pos, category: String): Some[(String, Move)] =
```

Comment:
returning a Some type doesn't match practise within SG/lila. Here you can return just the tuple as its is always a tuple and then decide what you want to do with it where ever it is called (e.g. Some(x), or even x.some)